src/compiler/parser.ts

```diff
@@ -6814,6 +6817,69 @@ namespace ts {
                     return finishNode(result);
                 }
 
+                function parseAuthorTag(start: number, tagName: Identifier): JSDocAuthorTag {
+                    const result = <JSDocAuthorTag>createNode(SyntaxKind.JSDocAuthorTag, start);
+                    result.tagName = tagName;
+
+                    const comment = tryParse(() => tryParseAuthorNameAndEmail());

```

Comment:
Since you aren't putting the author address into a separate field, I think would be pretty simple to make this work without backtracking.

Reply:
I'm not sure about the best way to achieve that. I return the parsing back to `parseTag` in case there is no email address provided instead of taking over parsing for the tag in all cases.